src/lib/components/Uploader/uploader.component.js

```diff
@@ -0,0 +1,236 @@
+import auth from "solid-auth-client";
+import { UploadedFiles, SolidError } from "@entities";
+
+type Props = {
+  fileBase: String,
+  limitFiles: number,
+  render: Node,
+  onComplete?: (files: Array<UploadedFiles>) => void,
+  onDrop?: (files: Array<UploadedFiles>) => void,
+  onError?: (error: SolidError) => void,
+  onStart?: () => void,
+};
+
+class Uploader extends Component<Props> {
+  counter: number;
+  positionFile: number;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      dragging: false,
+      files: [],
+      uploadedFiles: []
+    };
+    this.fileInput = React.createRef();
+    this.positionFile = 0;
+  }
+
+  componentDidMount() {
+    window.addEventListener("dragover", (event: React.EventHandler) => {
+      this.overrideEvent(event);
+    });
+    window.addEventListener("drop", (event: React.EventHandler) => {
+      this.overrideEvent(event);
+    });
+  }
+
+  componentDidUpdate(prevProps: Object, prevState: Object) {
+    // When files updates means that we have a new files to upload
+    if (this.state.files !== prevState.files && this.state.files.length > 0) {
+      this.upload();
+    }
+    // we check if all files was uploaded and fire onComplete props
+    if (
+      this.state.uploadedFiles.length > 0 &&
+      prevState.uploadedFiles !== this.state.uploadedFiles
+    ) {
+      this.onComplete(this.state.uploadedFiles);
+    }
+  }
+  /**
+   * Upload files to Solid Pod using fetch from solid-auth-client
+   * @params{Object} options
+   */
+  upload = async (options: Object) => {
+    const { fileBase, onError } = this.props;
+    const { files } = this.state;
+    let suffix = "";
+
+    // We read each file and upload to pod using Base64
+    files.forEach(file => {
+      const reader = new FileReader();
+
+      reader.onload = async f => {
+        try {
+          // Get image Base64 string
+          const data = f.target.result;
+          // Check if file has extension and add suffix string
+          if (file.type && file.type !== "") {
+            if (file.type !== lookup(file.name)) {
+              suffix = `_. ${extension(file.type)}`;
+            }
+          } else {
+            const error = {
+              type: "file",
+              statusText: "Unsupported Media Type",
+              code: 415
+            };
+            throw error;
+          }
+          // Get destination file url
+          const destinationUri = `${fileBase}/${encodeURIComponent(
+            file.name
+          )}${suffix}`;
+
+          // Send file on Base64 to server using fetch from solid-auth-client
+          const response = await auth.fetch(destinationUri, {
+            method: "PUT",
+            force: true,
+            headers: {
+              "content-type": file.type,
+              credentials: "include"
+            },
+            body: data
+          });
+          // Is all is fine we add new files into uploadedFiles array
```

Comment:
Typo, should read 

// If everything is fine, we add new files into the uploadedFiles array